### PR TITLE
Add auto-merge config

### LINKED ISSUE #183: Have roxbot deal with dependabot updates
Manually handling dependabot updates is annoying, and we tend to just blindly accept the updates anyway. Perhaps we should consider adding roxbot to this repo and having it take care of it for us. Thoughts @stackrox/maple ?

### PR DESCRIPTION
**Pull request by vikin91**
Fixes #183

### CODE REVIEW

File: .github/workflows/auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: auto-merge
+
+on:
+  pull_request_target:
```
Comment (RTann) [inline .github/workflows/auto-merge.yml:4]: is this needed or just here for testing?

File: .github/workflows/auto-merge.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+    if: github.actor == 'dependabot[bot]'
```
Comment (RTann) [inline .github/workflows/auto-merge.yml:9]: looks like stackrox/stackrox uses a slightly different condition (adds `auto-merge` flag to dependabot PRs). Do we want to do that here, too, to mimic that repo?

Comment (janisz) [reply .github/workflows/auto-merge.yml:9]: In stackrox we have label to exclude some PRs from automerge I think it's not required here the only problem with dependencies here is that upgrading them make break stackrox

**Review by janisz — APPROVED** [review]

File: .github/workflows/auto-merge.yml (hunk 1 above)
Comment (vikin91) [reply .github/workflows/auto-merge.yml:4]: I am not fully sure if this is needed. I followed suggestion from @janisz . I guess, we can merge and test it :)

**Comment by vikin91** [comment]
It looks, like the auto-merge is being skipped: https://github.com/stackrox/go-grpc-http1/actions/runs/8643824443?pr=188
Need to give it another look

**Pull request merged**